docs/user_manual/performance-guide.md

```diff
@@ -6,6 +6,57 @@ SPDX-License-Identifier: MPL-2.0
 
 # Guidelines for performance enhancement
 
+The `power-grid-model` is a library that shines in its ability to handle calculations at scale in.
+It remains performant, even when doing calculations with one or a combination of the following extremes (non-exhaustive):
+
+- Large grids
+- Batch calculations with many scenarios
+- Many changes in the grid in each scenario
+- Meshed grids
+
+To achieve that high performance, several optimizations to the code are made.
+To use those optimizations to their fullest extend, we recommend our users to follow the following guidelines.
+
+## Data validity
+
+Many of our optimizations rely on assuming input data validity and the fact that the provided grid is reasonably close to realistic.
+Non-convergence, underdetermined equations (sparse matrices) or other unexpected behavior may therefore be encountered when the data is not realistic.
+
+To keep the PGM performant, checks on hard physical bounds are offloaded to a separate tool, the [data validator](data-validator.md).
+However, these checks are extremely expensive and should therefore not be used in production environments at scale when performance matters.
+Instead, we recommend using the data validator specifically for debugging purposes.
+
+```{note}
+Some combinations of input data are not forbidden by physics, but still pose unrealistic conditions, e.g. a source with a very low short-circuit power, and unexpected behavior of the calculation core as a result.
+Vagueness and case-dependence make it hard to check what can be considered "unrealistic", and the [data validator](data-validator.md) will therefore not catch such cases.
```

Comment:
```suggestion
Vagueness and case-dependence make it hard to check what can be considered "unrealistic", and the [data validator](data-validator.md) will therefore not catch such cases. Most of the limits are bounded only at one end for example, extremes of which can result in unexpected behaviour.
```
How does this sound? The sentence has less consequence, so resolve if you dont agree.

Reply:
i wouldn't go for this one cause it may distract the reader from the actual goal of this page (performance, not correctness).

Reply:
slightly improved the note tho